ctest/README.md

```diff
@@ -0,0 +1,338 @@
+
+Enable/disable samples compilation.
+
+##### OPENCV_FEATURES_ONLY
+
+List of features, which should be enabled. All other features will be disabled.
+For example:
+
+```CMake
+set(OPENCV_FEATURES_ONLY TBB FFMPEG GTK)
+```
+
+##### OPENCV_FEATURES_ENABLE
+
+List of features, which should be enabled. The list will be combined with default settings.
+For example:
+
+```CMake
+set(OPENCV_FEATURES_ENABLE OPENGL)
+```
+
+##### OPENCV_FEATURES_DISABLE
+
+List of features, which should be disabled. The list will be combined with default settings.
+For example:
+
+```CMake
+set(OPENCV_FEATURES_DISABLE CUDA)
+```
+
+
+
+### 3.4. Optional testing settings
+
+##### CTEST_UPDATE_CMAKE_CACHE
+
+True, if the testing script should overwrite CMake cache on each launch.
+
+##### CTEST_EMPTY_BINARY_DIRECTORY
+
+True, if the testing script should clean build directory on each launch.
+
+##### CTEST_WITH_TESTS
+
+Enable/disable test launching.
+
+##### CTEST_TEST_TIMEOUT
+
+Timeout in seconds for single test execution.
+
+##### CTEST_WITH_MEMCHECK
+
+Enable/disable memory check analysis.
+
+##### CTEST_WITH_COVERAGE
+
+Enable/disable CTest-based code coverage analysis.
+
+##### CTEST_WITH_SUBMIT
+
+Enable/disable submission to remote server.
+
+##### CTEST_CMAKE_GENERATOR
+
+CMake generator.
+
+##### CTEST_CONFIGURATION_TYPE
+
+CMake configuration type (eg. Release, Debug).
+
+##### CTEST_BUILD_FLAGS
+
+Extra options for build command. For example:
+
+    set(CTEST_BUILD_FLAGS "-j7")
+
+--------------------------------------------------------------------------
+
+## 4. Usage from build tree
+
+The same testing script can be used from OpenCV build tree.
+
+To enable it, turn on `ENABLE_CTEST` option in CMake configuration.
+It will add new targets, which will call CTest tool with appropriate configuration:
+
+* `$ make Experimental`
+* `$ make Performance`
+
+--------------------------------------------------------------------------
+
+## 5. Usage with CI systems
+
+The testing script can be used with CI systems, like buildbot, Jenkins, etc.
+The CI system might call the same CTest command to perform project configuration, build and testing.
+
+The testing script supports step-by-step mode, to split all steps on CI system. For example:
+
+    $ ctest -VV -S ~/Dashboards/opencv/opencv_test.cmake,Start
+    $ ctest -VV -S ~/Dashboards/opencv/opencv_test.cmake,Configure
+    $ ctest -VV -S ~/Dashboards/opencv/opencv_test.cmake,Build
+    $ ctest -VV -S ~/Dashboards/opencv/opencv_test.cmake,Test
+    $ ctest -VV -S ~/Dashboards/opencv/opencv_test.cmake,Coverage
+    $ ctest -VV -S ~/Dashboards/opencv/opencv_test.cmake,MemCheck
+    $ ctest -VV -S ~/Dashboards/opencv/opencv_test.cmake,Submit
+    $ ctest -VV -S ~/Dashboards/opencv/opencv_test.cmake,Extra
+
+--------------------------------------------------------------------------
+
+[SCRIPT_LINK]: <https://raw.githubusercontent.com/jet47/opencv/ctest-dashboard-testing/ctest/opencv_test.cmake>
```

Comment:
Until the MR is merged, yes. The script file currently is available for download from my fork only.